### ISSUE TITLE
Nodes search: Searching for a node with corner case accounting records show empty

### ISSUE THREAD

**Issue by julsemaan** [inverse-inc/packetfence#2863]
As discussed verbally with @jrouzierinverse 

The faulty query (with obfuscated MAC):
```
SELECT `node`.`mac`, `node`.`pid`, `node`.`voip`, `node`.`bypass_vlan`, `node`.`status`, `node`.`category_id`,`node`.`bypass_role_id`, `node`.`user_agent`, `node`.`computername`, `node`.`last_arp`, `node`.`last_dhcp`, `node`.`notes`, IF(lastskip = '0000-00-00 00:00:00', '', lastskip) AS lastskip, IF(detect_date = '0000-00-00 00:00:00', '', detect_date) AS detect_date, IF(regdate = '0000-00-00 00:00:00', '', regdate) AS regdate, IF(unregdate = '0000-00-00 00:00:00', '', unregdate) AS unregdate, IF(last_seen = '0000-00-00 00:00:00', '', last_seen) AS last_seen, IFNULL(node_category.name, '') AS category, IFNULL(node_category_bypass_role.name, '') AS bypass_role, IFNULL(device_class, ' ') AS device_class, IFNULL(device_type, ' ') AS device_type, IFNULL(device_version, ' ') AS device_version, IF(r1.acctstarttime IS NULL,'unknown',IF(r1.acctstoptime IS NULL, 'on', 'off')) AS online, `ip4log`.`ip` AS last_ip, `locationlog`.`switch` AS switch_id, `locationlog`.`switch_ip` AS switch_ip, `locationlog`.`switch_mac` AS switch_mac, `locationlog`.`port` AS switch_port, `locationlog`.`ifDesc` AS switch_port_desc, `locationlog`.`ssid` AS last_ssid FROM `node` LEFT JOIN `node_category` ON `node_category`.`category_id` = `node`.`category_id` LEFT JOIN `node_category` AS node_category_bypass_role ON `node_category_bypass_role`.`category_id` = `node`.`bypass_role_id` LEFT JOIN `ip4log` ON `ip4log`.`ip` = ( SELECT `ip` FROM `ip4log` WHERE `mac` = `node`.`mac` ORDER BY `start_time` DESC LIMIT 1 ) LEFT JOIN `locationlog` ON `node`.`mac` = `locationlog`.`mac` AND `locationlog`.`end_time` = '0000-00-00 00:00:00' LEFT JOIN `locationlog` AS locationlog2 ON `node`.`mac` = `locationlog2`.`mac` AND `locationlog2`.`end_time` = '0000-00-00 00:00:00' AND ( `locationlog`.`start_time` < `locationlog2`.`start_time` OR ( `locationlog`.`start_time` = `locationlog2`.`start_time` AND `locationlog`.`id` < `locationlog2`.`id` ) OR `locationlog`.`start_time` IS NULL ) LEFT JOIN `radacct` AS r1 ON `node`.`mac` = `r1`.`callingstationid` LEFT JOIN `radacct` AS r2 ON `node`.`mac` = `r2`.`callingstationid` AND ( `r1`.`acctstarttime` < `r2`.`acctstarttime` OR ( `r1`.`acctstarttime` = `r2`.`acctstarttime` AND `r1`.`radacctid` < `r2`.`radacctid` ) OR `r1`.`acctstarttime` IS NULL ) WHERE ( `node`.`mac` = '00:11:22:33:44:55' ) AND ( `r2`.`radacctid` IS NULL AND `locationlog2`.`id` IS NULL )   ORDER BY `node`.`mac` ASC LIMIT 0 , 26; 
```

A dump was provided with the faulty data to @jrouzierinverse for investigation

Happens on 7.3

**Comment by jrouzierinverse** [comment]
This has been fixed on 7.4 and devel will backport fix to 7.3 and below

**Comment by julsemaan** [comment]
@jrouzierinverse 

Can you backport the fix so we close this
